Formula/amp.rb

```diff
@@ -0,0 +1,27 @@
+class Amp < Formula
+  desc "Text editor for your terminal"
+  homepage "https://amp.rs"
+  url "https://github.com/jmacdonald/amp/archive/0.3.2.tar.gz"
+  sha256 "9080c894111149d58a0b805e0c08edaf1d9e7cd56c4b955ba9111feec6b55208"
+  head "https://github.com/jmacdonald/amp.git"
+
+  depends_on "rust" => :build
+  depends_on "cmake" => :build
+  depends_on "openssl"
+  depends_on "libssh2"
+
+  def install
+    system "cargo", "build", "--release"
+    bin.install "target/release/amp"
+  end
+
+  test do
+    # Start and stop amp gracefully.
+    pid = spawn "#{bin}/amp"
```

Comment:
It'd be good to get a more substantive test here -- can `amp` be fed data via `expect` or some other tool?

Reply:
Inconveniently, `amp` doesn't have a headless/non-interactive mode at all. I'd thought about adding a `--version` argument to print the current version and exit, but that'd only really be to satisfy this test. 🙁

Reply:
For what it's worth, this test _should_ cover the most basic scenario (and the one that's biting people building from source): missing dynamically linked objects. Amp won't start at all if those are missing, and the exit code should be non-zero.

Reply:
Unfortunately, we couldn't accept a test with (just) `--version` -- something that tests a reasonably large chunk of the program's codebase is required.

My best advice in this case (since there's no headless mode to test) is to try `expect` out. Check out the `jupyter` and `tcpstat` formulae for examples.

Reply:
Makes perfect sense. I'll look into those and will update the PR with something more robust. Thanks for your help; it's appreciated. 🙂